### PR TITLE
use `::class` constant to reference class names

### PR DESCRIPTION
**Pull request by Tobion**
Based on #1251 and will be rebased once merged.

### CODE REVIEW

**Comment by ruflin** [comment]
@Tobion Can you rebase?

**Comment by Tobion** [comment]
Done

**Comment by ruflin** [comment]
That would be our first "hard" break with PHP 5.4, right?

**Comment by ruflin** [comment]
@Tobion Did you do all this changes manually or is there a tool that does this automatically?

**Comment by Tobion** [comment]
Semi-automated. I wrote a regex for most cases.

**Comment by Tobion** [comment]
Could be that this is the first change requiring php 5.5.

File: test/phpunit.xml.dist [hunk 1]
```diff
@@ -1,22 +1,19 @@
...
+<!-- https://phpunit.de/manual/current/en/appendixes.configuration.html -->
 <phpunit
-    backupGlobals="false"
-    backupStaticAttributes="false"
-    bootstrap="./bootstrap.php"
-    convertErrorsToExceptions="true"
-    convertNoticesToExceptions="true"
-    convertWarningsToExceptions="true"
-    processIsolation="false"
-    stopOnError="false"
```
Comment (ruflin) [inline test/phpunit.xml.dist]: @Tobion Were the once you removed not needed anymore?

Comment (Tobion) [inline test/phpunit.xml.dist]: They are the defaults which are usually not specified again.

Comment (ruflin) [inline test/phpunit.xml.dist]: I think I specified them a long time ago when I had issue with different phpunit version having different defaults.

File: test/phpunit.xml.dist [hunk 2]
```diff
@@ -1,22 +1,19 @@
...
-    stopOnSkipped="false"
-    syntaxCheck="false"
-    verbose="true"
-    >
+  xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+  xsi:noNamespaceSchemaLocation="http://schema.phpunit.de/5.7/phpunit.xsd"
+  backupGlobals                           = "false"
+  colors                                  = "true"
+  beStrictAboutTestsThatDoNotTestAnything = "true"
+  beStrictAboutOutputDuringTests          = "true"
```
Comment (ruflin) [inline test/phpunit.xml.dist]: Interesting names :-)

**Comment by ruflin** [comment]
@Tobion Thanks for the cleanup. Merged.

**Pull request merged**